File: Lab2/peak.cpp
#include <iostream>
using namespace std;

int getPeak(int arr[], int high, int low){
	
	int mid=(high+low)/2;
	if (mid==0)
		return 0;
	if (arr[mid]>arr[mid+1] && arr[mid]>arr[mid-1])
		return arr[mid];
	else if(arr[mid-1]<arr[mid])
		return getPeak(arr,mid+1,high);
	else 
		return getPeak(arr,low,mid-1);
	
}

int main(){
	cout<<"Enter the number of elements: ";
	int n;
	cin>>n;
	int arr[n];
	cout<<"Enter a sequence of integers that first strictly increase and then strictly decrease: ";
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	
	cout<<"\n"<<getPeak(arr,0,n)<<"\n";
}

File: Lab2/peak.py
def getPeak(arr):
	n=len(arr)
	mid=n//2
	if len(arr)==1:
		print(arr[0])
	elif arr[mid]>arr[mid+1] and arr[mid]>arr[mid-1]:
		print( arr[mid])
	elif arr[mid-1]<arr[mid]:
		return getPeak(arr[n//2:])
	else:
		return getPeak(arr[:n//2])


def main():
	print("Enter a sequence of integers that first strictly increase and then strictly decrease: ")
	arr=[int(_) for _ in input().split()]
	getPeak(arr)

if __name__ == '__main__':
	main()